tools/scoring/score/dimensions/dimension.py

```diff
@@ -11,71 +11,104 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-"""Core functionality parent class"""
+"""Core component base class"""
 
-from dataclasses import dataclass
+from typing import Any, Dict, List, Tuple
+from validate.entity_instance import EntityInstance
 
 
-@dataclass
 class Dimension:
   """
     Container for floating-point results which
     quantify success in each dimension
 
     Attributes:
-      correct_virtual: Number of attempts achieved within virtual devices
-      correct_reporting: Number of attempts achieved within reporting devices
-      correct_ceiling_virtual: Number of attempts possible to achieve
+      translations: Proposed and solution translations
+        for all matched reporting entities. Assigned via argument
+      deserialized_files: Parsed configuration files.  Assigned via argument
+
+      correct_virtual: Number of successful attempts within virtual devices
+      correct_reporting: Number of successful attempts within reporting devices
+      correct_ceiling_virtual: Number of attempts possible
         within virtual devices
-      correct_ceiling_reporting: Number of attempts possible to achieve
-        within reporting devices
-      incorrect_virtual: Number of attempts not achieved within virtual devices
-      incorrect_reporting: Number of attempts not achieved
+      correct_ceiling_reporting: Number of attempts possible
         within reporting devices
+      incorrect_virtual: Number of failed attempts within virtual devices
+      incorrect_reporting: Number of failed attempts within reporting devices
 
     Properties:
       result_composite: Calculated result for all devices
       result_virtual: Calculated result for virtual devices
       result_reporting: Calculated result for reporting devices
   """
-  correct_virtual: int
-  correct_reporting: int
-  correct_ceiling_virtual: int
-  correct_ceiling_reporting: int
-  incorrect_virtual: int
-  incorrect_reporting: int
+  def __init__(self,
+               *,
+               translations: Dict[str, List[Tuple[str, Any]]] = None,
+               deserialized_files: Dict[str, Dict[str, EntityInstance]] = None):
```

Comment:
what are those `str` keys can you please add more info in the docstring. for both the translation and the deserialized files

Reply:
They're `cloud_device_id`s… I'll make types.

Reply:
Dict[str, Dict[str, EntityInstance]]

there are two str?